### ISSUE TITLE
MultiDataSetIteratorSplitter throws NullPointerException

### ISSUE THREAD

**Issue by wakawaka54** [deeplearning4j/deeplearning4j#9086]
MultiDataSetIteratorSplitter throws NullPointerException?

https://github.com/eclipse/deeplearning4j/blob/1de8837c39c368b46e3943a2d24f92cfcff5231e/deeplearning4j/deeplearning4j-data/deeplearning4j-utility-iterators/src/main/java/org/deeplearning4j/datasets/iterator/MultiDataSetIteratorSplitter.java#L104-L107

**Comment by alexanderst** [comment]
Yes, line 107. This constructor is not covered with unit test, so the bug was missed.

**Comment by agibsonccc** [comment]
Closing — this issue is over 2 years old with no recent activity or reproduction steps. The underlying code has been significantly rewritten. If you still experience this issue with a recent version, please open a new issue.
